### PR TITLE
Add Curving Road

### PR DESCRIPTION
**Pull request by Seren-35**
Source: http://inabapuzzle.com/honkaku/curving.html

In edit mode, I made circle placement follow the same behavior as shaded cells in dynasty genres, since two circles can never be adjacent in a solvable puzzle, whereas circles one empty cell apart seem to be a common pattern in puzzles I've seen.

When checking for paths that turn only once, at most one erroneous path per circle is highlighted. There can be quadratically many of those in total so highlighting all of them in sufficiently wrong answers can easily cover most of the grid and turn it unreadable. This seemed like a good compromise.

### CODE REVIEW

**Comment by Seren-35** [comment]
Apparently CI actually got mad at me for CR LF line endings this time. No clue why since I haven't done anything differently.

File: test/script/kabingurodo.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		[
+			"kabinOneTurn",
+			"pzprv3/kabingurodo/6/6/1 . . . . . /. . 1 . 1 . /. . . . . . /1 . . . . . /. . 1 . . 1 /. . . 1 . . /. . # . . # /# . . # . . /. . . . # . /. . # . . . /# . . # . . /. . # . . # /"
+		],
+		[
+			null,
+			"pzprv3/kabingurodo/6/6/1 . . . . . /. . 1 . 1 . /. . . . . . /1 . . . . . /. . 1 . . 1 /. . . 1 . . /. + # + + + /# + . # . # /+ + + + + + /. + # + # + /# + . # + . /+ + # . + # /"
+		]
+	],
+	inputs: []
```
Comment (x-sheep) [inline test/script/kabingurodo.js:35]: The functions in `MouseEvent` are not covered by tests. Could you add a few input tests for the Editing part here?

**Review by x-sheep — CHANGES_REQUESTED** [review]
Thanks for the contribution! I like how the error highlighting looks.

**Comment by Seren-35** [comment]
Just occurred to me I should also test what happens when placing a circle on top of a shaded cell. This apparently throws an exception but, as best as I can tell, not because of my code - the combination of `numberRemainsUnshaded: true` and `edit: ["circle-unshade"]` seems to just do this.

**Review by x-sheep — APPROVED** [review]
Thanks for doing the changes. I'm going to release San-Anko on monday, and this genre 2 weeks after.

**Pull request merged**